aten/src/ATen/ParallelCommon.cpp

```diff
@@ -0,0 +1,56 @@
+#include <ATen/Parallel.h>
+
+#include <ATen/Config.h>
+#include <ATen/Version.h>
+
+#include <sstream>
+#include <thread>
+
+#ifdef TH_BLAS_MKL
+#include <mkl.h>
+#endif
+
+#ifdef _OPENMP
```

Comment:
Does this use the build settings you created in an earlier PR to determine whether openmp is available or not? Maybe it makes sense to explicitly prevent us from including openmp, even if it is technically available as indicated by this macro.

Reply:
I removed most of `_OPENMP` macro usage from PyTorch, except for a few places, including this - here it is just for the code below that prints information about libraries used, etc.

Reply:
Is there an advantage to / necessity for using _OPENMP?

Reply:
I removed most of _OPENMP checks before because we expect the code to be agnostic of backend and just call ATen/Parallel function.

Here we include it because we call `omp_get_max_threads` explicitly below - only because I'd want to output OpenMP settings in `get_parallel_info`

Reply:
So, no there's no advantage, it's the opposite - we'd want to not have them

Reply:
@ezyang suggested adding a lint rule https://github.com/pytorch/pytorch/issues/20466